backend/users/admin.py

```diff
@@ -29,15 +44,23 @@
 )
 
 
-def create_users(user_type, records):
+def create_users(user_type, records, request):
     headers = CSV_HEADERS[user_type]
     if set(records.fieldnames) != set(headers):
         raise Exception(f"Invalid CSV headers/columns. Expected: {headers}")
     for record in records:
-        if user_type == Faculty:
+        record = {key: (None if value == '' else value) for key, value in record.items()}
+        if user_type in (Faculty, ResearchScholar):
             record['dept'] = Department.objects.get(name=record['dept'])
+        if user_type == ResearchScholar:
+            supervisor = Faculty.objects.filter(name=record['supervisor'])
+            record['supervisor'] = supervisor.first()
         # TODO: Update object if it already exists instead of failing hard
-        user_type.objects.create(**record)
+        try:
+            user_type.objects.create(**record)
+        except Exception as err:
```

Comment:
Why catch the error here? It will automatically be raised to the surrounding function and be caught and handled there (along with proper message)

Reply:
for row number of error. surrounding function can't show that.
@iamkroot

Reply:
We can't send the entire request object just to show the line number. The function should only care about creating the objects from given records. In case there's an error, _raise_ it (along with proper information) and let the calling function handle that.